.github/workflows/deploy.yml

```diff
@@ -12,98 +12,50 @@ on:
     types:
       - created
 
-env:
-  GKE_REGION: us-west1
-  GKE_CLUSTER: main
-  CONTAINER_REGISTRY: us-west1-docker.pkg.dev
-
 jobs:
   deploy:
     if: github.event_name == 'release' || github.ref == 'refs/heads/main' || contains(github.event.head_commit.message, '[stagingdeploy]') || contains(github.event.head_commit.message, '[testnetdeploy]')
     name: Deploy
     runs-on: ubuntu-latest
     steps:
-      - name: Dump GitHub context
-        env:
-          GITHUB_CONTEXT: ${{ toJSON(github) }}
-        run: echo "$GITHUB_CONTEXT"
-
-      - name: Install Go
-        uses: actions/setup-go@v2
-        with:
-          go-version: v1.17
-
       - name: Checkout
         uses: actions/checkout@v2
 
       - name: Prepare
+        shell: bash
         id: prep
         run: |
           if [[ ${{ github.ref }} == refs/heads/main ]] || [[ "${{github.event.head_commit.message}}" == *"[testnetdeploy]"* ]]; then
-            echo "::set-output name=deployment::testnet"
+            echo "::set-output name=deployment::testnet" 
           else
             echo "::set-output name=deployment::staging"
+            echo "::set-output name=vm_host::${{ secrets.STAGING_VM_HOST }}"
           fi
-          echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
-
-      - name: Set up Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@v1
 
-      - name: Cache Docker layers
-        uses: actions/cache@v2
-        with:
-          path: /tmp/.buildx-cache
-          key: ${{ runner.os }}-buildx-${{ github.sha }}
-          restore-keys: |
-            ${{ runner.os }}-buildx-
+          # This is the official way to setup secret values that are multi-line, such as an SSH private key.
+          echo 'vm_ssh_key<<EOF' >> $GITHUB_ENV
+          echo "${{ secrets.VM_SSH_KEY }}" >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
```

Comment:
In this `Prepare` step, we setup two new outputs:
- `vm_host` which will have the VM IP that we'll connect via SSH
- `vm_ssh_key` which contains the SSH keys to connect to the VM.

Note that `vm_host` is conditional depending on the usual commit substring `[<env>deploy]` idea or branch.
The `vm_ssh_key` is independent since all the VMs are configured to accept connections from a single SSH key that we use for GH actions.

Note that `vm_ssh_key` is a multiline value, so we need to do what's done in L35 to set up properly.